lib/plugins/aws/deploy/compile/events/stream/index.js

```diff
@@ -78,8 +88,15 @@ class AwsCompileStreamEvents {
                 .Error(errorMessage);
             }
 
-            const streamType = EventSourceArn.split(':')[2];
-            const streamName = EventSourceArn.split('/')[1];
+            const streamType = event.stream.type || EventSourceArn.split(':')[2];
+            const streamName = (function () {
+              if (EventSourceArn['Fn::GetAtt']) {
+                return EventSourceArn['Fn::GetAtt'][0];
+              } else if (EventSourceArn['Fn::ImportValue']) {
+                return EventSourceArn['Fn::ImportValue'];
```

Comment:
That's fair. I'm not sure what I'd use instead as the name. Maybe force-truncating it at some (reasonable) size in the `provider.naming` library so it won't return 255+ chars?